src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -94,8 +118,49 @@ public VisualStudioProjectTracker(IServiceProvider serviceProvider)
             }
         }
 
+        private void WaitForScheduledForegroundAffinitizedActions()
+        {
+            AssertIsForeground();
+
+            lock (_gate)
+            {
+                _taskForForegroundAffinitizedActions.Wait();
```

Comment:
I admit I didn't understand why this mechanism existed at all. If you think it's unnecessary then let's delete it, but clearly you thought it was worth adding in the first place!